src/context/context.ts

```diff
@@ -458,10 +460,30 @@ export class WorkflowContext<TInitialPayload = unknown> {
     }
   }
 
+  public async invoke<TInitialPayload, TResult>(
+    stepName: string,
+    settings: LazyInvokeStepParams<TInitialPayload, TResult> & { waitForResult: false }
+  ): Promise<CallResponse<TResult>>;
+  public async invoke<TInitialPayload, TResult>(
+    stepName: string,
+    settings: LazyInvokeStepParams<TInitialPayload, TResult> & { waitForResult?: true }
+  ): Promise<InvokeStepResponse<TResult>>;
   public async invoke<TInitialPayload, TResult>(
     stepName: string,
     settings: LazyInvokeStepParams<TInitialPayload, TResult>
-  ) {
+  ): Promise<CallResponse<TResult> | InvokeStepResponse<TResult>> {
+    const { waitForResult } = settings;
+    if (!waitForResult) {
+      return await this.call(stepName, {
```

Comment:
when we use call, step types will appear as call, even though the method is invoke